src/encoder.js

```diff
@@ -388,62 +419,79 @@ export default class Encoder extends Extendable {
       target.push(tags.BINARY_);
       this.encodeValue(value.byteLength, target);
       this.pushArrayElements(new Uint8Array(value), target);
-    } else if (Array.isArray(value)) {
-      let numElements = value.length;
+    } else {
+      if (this.remainingDepth !== null) {
+        if (this.remainingDepth === 0) {
```

Comment:
I prefer bounds checks like this:

```suggestion
        if (this.remainingDepth <= 0) {
```

Reply:
/shrug sure I guess